addons/website_slides_survey/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -1,15 +1,39 @@
 odoo.define('website_slides_survey.fullscreen', function (require) {
     "use strict";
-    
+
     var core = require('web.core');
-    var _t = core._t;
+    var QWeb = core.qweb;
     var Fullscreen = require('website_slides.fullscreen');
-    
+
     Fullscreen.include({
         xmlDependencies: (Fullscreen.prototype.xmlDependencies || []).concat(
             ["/website_slides_survey/static/src/xml/website_slides_fullscreen.xml"]
         ),
+        _preprocessSlideData: function (slidesDataList){
+            slidesDataList = this._super.apply(this, arguments);
+            slidesDataList.forEach(function (slideData, index){
+                if (slideData.type === "certification"){
+                    slideData.certificationUrl = '/slides_survey/certification_url/get?slide_id=' + slideData.id;
+                }
+            });
+            return slidesDataList;
+        },
+        /**
+         * Extend the _renderSlide method so that slides of type "certification"
+         * are also taken into account and rendered correctly
+         *
+         * @private
+         * @override
+         */
+        _renderSlide: function (){
+            var def = this._super.apply(this, arguments);
```

Comment:
I think super should be called first since it removes the content from o_wslides_fs_content.
So this would be: 

```
return this._super.apply(this, arguments).then(function () {
    ...
   return $.when();
});
```

But I have to say I'm not entirely sure.

Reply:
Not here, as we want to get the deferred of the rendering. They are exclusif, and should be chained. e.i.: we don't need to wait the rendering of the quiz slide to check if it's a certification.